src/pretalx/mail/signals.py

```diff
@@ -8,6 +8,18 @@
 As with all event-plugin signals, the ``sender`` keyword argument will contain the event.
 """
 
+queuedmail_pre_send = EventPluginSignal()
+"""
+This signal is sent out before a ``QueuedMail`` will been sent.
+Receivers may set the ``sent`` timestamp to skip sending via the regular
+email backend but shall not alter any other data of the ``QueuedMail``
+instance.
+
+As with all event-plugin signals, the ``sender`` keyword argument will
+contain the event. Additionally, the ``mail`` keyword argument contains
+the ``QueuedMail`` instance itself.
```

Comment:
One more thought: We should figure out how to deal with exceptions raised by receivers of this signal.

Two real options:

- ignore exceptions – use the `send_robust` sending method, and document that we will disregard exceptions
- allow exceptions to abort the sending of emails: this is nicer for plugin authors, but at the same time, we don’t have a channel back to the user in most cases, as emails are sent asynchronously.

Since a plugin blocking the sending of an email has no way to tell the user **that** and **why** it has blocked the sending, I’m leaning towards the first option.

Reply:
I’m also leaning towards the first option. I thought about this a little bit already and didn't really have a better idea.